src/main/java/com/epam/spring/homework2/beans/ValidateFieldsAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,45 @@
+package com.epam.spring.homework2.beans;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+
+import java.lang.reflect.Field;
+
+public class ValidateFieldsAnnotationBeanPostProcessor implements BeanPostProcessor {
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        return bean;
+    }
+
+    @Override
+    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+        if (bean.getClass().isAnnotationPresent(ValidateFields.class)) {
+            try {
+                Field nameField = bean.getClass().getDeclaredField("name");
+                Field valueField = bean.getClass().getDeclaredField("value");
+
+                nameField.setAccessible(true);
+                valueField.setAccessible(true);
+
+                String name = (String) nameField.get(bean);
+                Integer value = (Integer) valueField.get(bean);
+
+                if (name == null) {
+                    System.out.println(bean.getClass().getSimpleName() + " is not valid. " + " Name can not be null");
+
+                } else if (value <= 0) {
+                    System.out.println(bean.getClass().getSimpleName() + " is not valid. " + " Value must be greater than 0");
+
+                } else {
+                    System.out.println(bean.getClass().getSimpleName() + " is valid!)");
+                }
```

Comment:
(nit, optional): There could be cases when name could be null and value would be negative, but using if/else if statements here would validate only one thing at a time. You can make a validator class with methods like `boolean isNamePresent(String name) `and `boolean isValuePositive(Integer value)` and call them here, that way we would have logs if validation is not successful per each field, and a return value to apply some logic if validation fails\passes.